File: java/src/test/java/problems1/ThreeSumTest.java
package problems1;

import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ThreeSumTest extends SampledTest {
    @Override
    public String getFileName() {
        return "three_sum.json";
    }

    @Test
    public void test() {
        for (ThreeSum solution : ThreeSum.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.threeSum(sample.input.nums))
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .map(ints -> Arrays.stream(ints).boxed().collect(Collectors.toList()))
                            .collect(Collectors.toList())
                    );
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, int[][]> {
        public static class Input {
            public int[] nums;
        }
    }
}

File: javascript/__tests__/problems1_100/three-sum.test.js
import {deepEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/three-sum';
import getSamples from '../../../testing/js/src/sample/samples';

describe(
    'ThreeSum',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach(sample =>
                        deepEqual(
                            solution.threeSum(sample.input.nums),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
